src/passes/externalContractHandler/externalContractInterfaceInserter.ts

```diff
@@ -80,12 +80,8 @@ function importExternalContract(
   contractInterfaces: Map<number, ContractDefinition>,
   ast: AST,
 ) {
-  const contractSourceUnit = contract.getClosestParentByType(SourceUnit);
-
   assert(sourceUnit !== undefined, 'Trying to import a definition into an unknown source unit');
 
-  if (contractSourceUnit === undefined || sourceUnit === contractSourceUnit) return;
-
   if (contract.kind === ContractKind.Library) return;
   if (contractInterfaces.has(contract.id)) return;
   contractInterfaces.set(contract.id, genContractInterface(contract, sourceUnit, ast));
```

Comment:
unrelated to your changes but on line 115 we only filter constructors from the interface, we should also filter not externally visible functions